src/cpp/rtps/messages/MessageReceiver.cpp

```diff
@@ -317,53 +316,63 @@ void MessageReceiver::processCDRMsg(
         const Locator_t& reception_locator,
         CDRMessage_t* msg)
 {
-    if (msg->length < RTPSMESSAGE_HEADER_SIZE)
-    {
-        logWarning(RTPS_MSG_IN, IDSTRING "Received message too short, ignoring");
-        return;
-    }
-
-    reset();
 
 #ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
     GuidPrefix_t participantGuidPrefix;
 #else
     GuidPrefix_t participantGuidPrefix = participant_->getGuid().guidPrefix;
 #endif // ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
-    dest_guid_prefix_ = participantGuidPrefix;
 
-    msg->pos = 0; //Start reading at 0
+#if HAVE_SECURITY
+    security::SecurityManager& security = participant_->security_manager();
+    CDRMessage_t* auxiliary_buffer = &crypto_msg_;
+    int decode_ret = 0;
+#endif // if HAVE_SECURITY
 
-    //Once everything is set, the reading begins:
-    if (!checkRTPSHeader(msg))
     {
-        return;
-    }
+        std::lock_guard<eprosima::shared_mutex> guard(mtx_);
 
-    notify_network_statistics(source_locator, reception_locator, msg);
+        if (msg->length < RTPSMESSAGE_HEADER_SIZE)
+        {
+            logWarning(RTPS_MSG_IN, IDSTRING "Received message too short, ignoring");
+            return;
+        }
```

Comment:
I think this can be kept at the beginning of the method. As it only depends on input, and not on the state of the `MessageReceiver`, there is no need for it to be protected by the mutex.